OTLP Logs Intake Endpoint (#26695)
DataDog/documentation: content/en/opentelemetry/otlp_logs.md

---
title: Datadog OTLP Logs Intake Endpoint
private: true
further_reading:
  - link: "https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/"
    tag: "External Site"
    text: "General OpenTelemetry SDK Configuration"
  - link: "https://opentelemetry.io/docs/reference/specification/sdk-environment-variables/"
    tag: "External Site"
    text: "OpenTelemetry Environment Variable Spec"
  - link: "https://opentelemetry.io/docs/reference/specification/protocol/exporter/"
    tag: "External Site"
    text: "OpenTelemetry Protocol Exporter"
---

{{< site-region region="ap1,gov" >}}
<div class="alert alert-warning">Datadog OTLP logs intake endpoint is not supported for your selected <a href="/getting_started/site">Datadog site</a> ({{< region-param key="dd_site_name" >}}).</div>
{{< /site-region >}}

## Overview

Datadog's OpenTelemetry Protocol (OTLP) logs intake API endpoint allows you to send logs directly to Datadog. With this feature, you don't need to run the [Datadog Agent][2] or [OpenTelemetry Collector + Datadog Exporter][1].

> You might prefer this option if you're looking for a straightforward setup and want to send logs directly to Datadog without using the Datadog Agent or OpenTelemetry Collector.

Reviewer: ```suggestion
Choose this option for a straightforward setup to send logs directly to Datadog without using the Datadog Agent or OpenTelemetry Collector.
```

## Configuration

To send OTLP data to the Datadog OTLP logs intake endpoint, you must configure the OTLP HTTP Protobuf exporter. The process differs depending on whether you are using automatic or manual instrumentation for OpenTelemetry.

<div class="alert alert-info">Based on your <a href="/getting_started/site/">Datadog site</a>, which is {{< region-param key=dd_datacenter code="true" >}}: Replace <code>${YOUR_ENDPOINT}</code> with {{< region-param key="otlp_logs_endpoint" code="true" >}} in the following examples.</div>

#### Automatic instrumentation

If you are using [OpenTelemetry automatic instrumentation][3], set the following environment variables:

```shell
export OTEL_EXPORTER_OTLP_LOGS_PROTOCOL="http/protobuf"
export OTEL_EXPORTER_OTLP_LOGS_ENDPOINT="{{< region-param key="otlp_logs_endpoint" >}}"
export OTEL_EXPORTER_OTLP_LOGS_HEADERS="dd-protocol=otlp,dd-api-key=${DD_API_KEY}"
```

#### Manual instrumentation

> If you are using manual instrumentation with OpenTelemetry SDKs, configure the OTLP HTTP Protobuf exporter programmatically using the examples below.

Reviewer: ```suggestion
If you are using manual instrumentation with OpenTelemetry SDKs, configure the OTLP HTTP Protobuf exporter programmatically using the following examples.
```
